okhttp-tests/src/test/java/okhttp3/MultipartBodyTest.java

```diff
@@ -131,27 +131,66 @@
                     RequestBody.create(
                         MediaType.get("text/plain"), "... contents of file1.txt ..."))
                 .addPart(
-                    Headers.of(
-                        "Content-Disposition", "file; filename=\"file2.gif\"",
-                        "Content-Transfer-Encoding", "binary"),
+                    new Headers.Builder()
+                        .addUnsafeNonAscii("Content-Disposition", "file; filename=\"file2€.gif\"")
```

Comment:
Can you move this out into a separate targeted test and leave the existing test unchanged?  The new test can be shorter and just focus on length and filename encoding.

Reply:
Yeah, I undid the change. Do you think I can even remove `expected` to do [assertEquals(expected, buffer.readUtf8())](https://github.com/square/okhttp/pull/4576/commits/a1182fc95d61fdefc9418afcb11729e6671eb439#diff-c0fbf53145399b63adc71321a0d0b864R190)?